aeweb.js

```diff
@@ -1,129 +1,212 @@
+
+            archethic.sendTransaction(transaction, argv.endpoint).then((response) => {
+                if (response.status == 'ok') {
+                    console.log(chalk.blue("Transaction Sent Successfully !"))
+                    console.log(chalk.green(argv.endpoint + "/api/last_transaction/" + (toHex(transaction.address)) + "/content?mime=" + mime.getType(argv.file)))
+                } else {
+                    console.log(chalk.red("Transaction not deployed ! Please check if funds are transferred successfully to the generated address"))
+                }
+            })
+        })
+    }
+})
+
+yargs.command({
+    command: 'deploy-folder',
+    describe: 'Deploy all files inside folder on-chain',
+    builder: {
+        seed: {
+            describe: 'Seed',
+            demandOption: true, // Required
+            type: 'string'
         },
 
         endpoint: {
             describe: 'Node Endpoint',
-            demandOption: true,  // Required
-            type: 'string'     
+            demandOption: true, // Required
+            type: 'string'
         },
 
-        path: {
-            describe: 'Path',
-            demandOption: true,  // Required
-            type: 'string'     
+        folder: {
+            describe: 'Folder Name',
+            demandOption: true, // Required
+            type: 'string'
+        }
+    },
+
+    handler: function (argv) {
+
+        function ReadDirectory(Directory) {
+            try {
+                fs.readdirSync(Directory).forEach(File => {
+                    const abs = path.join(Directory, File);
+                    if (fs.statSync(abs).isDirectory()) return ReadDirectory(abs);
+                    else return Files.push(abs);
+                });
+            } catch (e) {
+                console.error(e.message)
+            }
         }
 
+        ReadDirectory(argv.folder);
 
-    },
+        for (let i = 0; i < Files.length; i++) {
+            const hmac = crypto.createHmac(algo, argv.seed);
+            hmac.update(Files[i])
+            const seed = hmac.digest('hex')
+            Seed.push(seed)
+            const address = archethic.deriveAddress(seed, 0)
+            Address.push(address)
+        }
+
+        tx = archethic.newTransactionBuilder("transfer")
 
-    handler: function (argv)
-    {   function toHex(bytes) {
-                return bytes.reduce((str, byte) => str + byte.toString(16).padStart(2, '0'), '');
+        for (let i = 0; i < Address.length; i++) {
+            tx.addUCOTransfer(Address[i], 1.0)
         }
-        
-        const originPrivateKey = "01009280BDB84B8F8AEDBA205FE3552689964A5626EE2C60AA10E3BF22A91A036009"
-        let transaction
-
-        
-        const content = fs.readFileSync(argv.path)
-        
-        
-        
-      
+
+        txn = tx
+            .build(argv.seed, 0)
+            .originSign(originPrivateKey)
+
+        archethic.sendTransaction(txn, argv.endpoint)
```

Comment:
You should wait the completion before to continue further. Othewise the transaction may not be validated yet